design/one-pager-merge-jets-proposal.md

```diff
@@ -0,0 +1,370 @@
+based on ARM and Google is using Magic Modules and DCL to generate TF provider,
+so it's likely that they will choose to do that for their Crossplane provider as
+well. [Provider Strategy doc][strategy-doc] goes into more detail about these
+tools. In summary, Terrajet will be a temporary solution that we put in place to
+get more coverage and at some point we'll need to discuss how to change the
+implementation to those tools.
+
+With that in mind, it's a hard choice to invest in separate Jet-based providers
+knowing that they will, even though after years, be deprecated at one point and
+users will have to migrate. At the same time, classic providers are very far
+from having the same coverage level. This creates confusion and reluctance to
+use either of the providers, which also results in decreased pace of
+contributions because of the divergence of efforts.
+
+With a single mixed provider, the story is much clearer; we fill the gaps with
+Terrajet and over time the implementation of the resources will change to be
+more _native_; be it cloud vendors using their own approaches or handwritten API
+calls community writes that may be necessary to overcome a bug in TF provider.
+As long as the CRD schema and the API behavior is same, the maintainers can make
+the best choice depending on the situation, communicate any breaking changes and
+provide a smooth experience to the users. In case cloud vendor steps in and
+decides to start its own provider, then the story is still clear; there is a new
+provider maintained by its vendor and you should migrate to that one once you
+see fit as opposed to Jet vs classic choice where you need to think about
+different trade-offs with different and uncertain timelines.
+
+### Ownership by The Cloud Vendors
+
+As stated above, we hope that cloud vendors will come in and maintain their
+Crossplane providers. It seems like Terraform has achieved this goal only for
+GCP; Azure and AWS Terraform providers are not maintained by the vendors. In our
+case, Crossplane is a CNCF project with an open governance model, so we can't
+really compare it with Terraform, a property of HashiCorp, in that respect but
+it still provides a data point.
+
+The strategy we choose should take the following possibilities into account:
+* They may choose to bootstrap their own Crossplane provider, maybe even not
+under Crossplane organization.
+* They might decide to convert the underlying implementation to their own
+technology, similar to GCP Terraform provider that is [undergoing a
+process][gcp-tf-contributing] where they change implementation from handwritten
+API calls to Magic Modules and DCL.
+* They would provide the tools but not take part in maintenance of the codebase.
+
+In the first case, regardless of whether we go with separate or mixed providers,
+we will likely suggest users to migrate to the vendor-maintained provider so
+that they can get commercial grade support for the bugs in the provider. One
```

Comment:
Who is giving commercial grade support here? The vendor? I'm not sure we know enough to presume that will happen.

Reply:
I am assuming a level of support similar to their SDKs. Do you think no such support exists?

Reply:
I would not assume it does - it may be worth looking to open source projects like ASO and ACK to determine whether vendors deem those to be "commercially supported", which to me implies it's possible to pay for support for those tools.